libraries/AP_SpdHgtControl/AP_SpdHgtControl_Heli.cpp

```diff
@@ -0,0 +1,182 @@
+#include "AP_SpdHgtControl_Heli.h"
+#include <AP_HAL/AP_HAL.h>
+
+extern const AP_HAL::HAL& hal;
+
+// table of user settable parameters
+const AP_Param::GroupInfo AP_SpdHgtControl_Heli::var_info[] = {
+
+    // @Param: VEL_P
+    // @DisplayName: Velocity (horizontal) P gain
+    // @Description: Velocity (horizontal) P gain.  Determines the propotion of the target acceleration based on the velocity error.
+    // @Range: 0.1 6.0
+    // @Increment: 0.1
+    // @User: Advanced
+
+    // @Param: VEL_I
+    // @DisplayName: Velocity (horizontal) I gain
+    // @Description: Velocity (horizontal) I gain.  Corrects long-term difference in desired velocity to a target acceleration
+    // @Range: 0.02 1.00
+    // @Increment: 0.01
+    // @User: Advanced
+
+    // @Param: VEL_D
+    // @DisplayName: Velocity (horizontal) D gain
+    // @Description: Not used. Leave as zero.
+    // @User: Advanced
+
+    // @Param: VEL_FF
+    // @DisplayName: Velocity (horizontal) feed forward
+    // @Description: Velocity (horizontal) input filter.  Corrects the target acceleration proportionally to the desired velocity.
+    // @Range: 0 1
+    // @Increment: 0.01
+    // @User: Advanced
+
+    // @Param: VEL_IMAX
+    // @DisplayName: Velocity (horizontal) integrator maximum
+    // @Description: Velocity (horizontal) integrator maximum.  Constrains the I term contribution to acceleration target.
+    // @Range: 0 4500
+    // @Increment: 10
+    // @Units: cm/s/s
+    // @User: Advanced
+
+    // @Param: VEL_FLTT
+    // @DisplayName: Velocity (horizontal) target filter
+    // @Description: Velocity (horizontal) target filter.  Limits the maximum rate that the velocity target can be changed.
+    // @Range: 0 100
+    // @Units: Hz
+    // @User: Advanced
+
+    // @Param: VEL_FLTE
+    // @DisplayName: Velocity (horizontal) error filter
+    // @Description: Velocity (horizontal) error filter.  Limits the maximum rate that the velocity error can be changed.
+    // @Range: 0 100
+    // @Units: Hz
+    // @User: Advanced
+
+    // @Param: VEL_FLTD
+    // @DisplayName: Velocity (horizontal) D filter
+    // @Description: No D term in this controller.  Not used.  Leave as 0.
+    // @Units: Hz
+    // @User: Advanced
+    AP_SUBGROUPINFO(_pid_vel, "VEL_", 12, AP_SpdHgtControl_Heli, AC_PID),
```

Comment:
this should be "1" instead of a "12" as shown below.
AP_SUBGROUPINFO(_pid_vel, "VEL_", 1, AP_SpdHgtControl_Heli, AC_PID),

Reply:
Done